src/components/TaskItem.tsx

```diff
@@ -0,0 +1,71 @@
+"use client";
+import { useState } from "react";
+import Button from "./ui/Button";
+import TextField from "./ui/TextField";
+import EditTaskModal from "./EditTaskModal";
+import {
+  useUpdateTaskMutation,
+  useDeleteTaskMutation,
+} from "@/redux/features/tasks/tasksAPI";
+
+type Task = {
+  id: string;
+  title: string;
+  completed: boolean;
+};
+
+interface TaskItemProps {
+  task: Task;
+}
+
+export default function TaskItem({ task }: TaskItemProps) {
+  const [editId, setEditId] = useState<string | null>(null);
+  const [updateTask] = useUpdateTaskMutation();
```

Comment:
please get isLoading state to show the progress of PUT/POST API loading in component.
```suggestion
  const [updateTask, isLoading: isLoadingUpdateTask] = useUpdateTaskMutation();
```